protosocket-connection/src/connection.rs

```diff
@@ -56,8 +58,26 @@ impl<Bindings: ConnectionBindings> Unpin for Connection<Bindings> {}
 impl<Bindings: ConnectionBindings> Future for Connection<Bindings> {
     type Output = ();
 
+    /**
+     * The poll method is the core of the Connection's asynchronous behavior.
+     * It's recommended to familiarize yourself with the ConnectionBindings trait
+     * in the types.rs file of this crate before proceeding.
+     *
+     * This method performs the following steps:
+     *
+     * 1. Check for read readiness on the Connection's TcpStream.
+     * 2. If data is available, read raw bytes into the receive_buffer (up to max_buffer_length).
```

Comment:
For symmetry with the last line these cases should be collapsed
```suggestion
     * 1. Check for read readiness and read into the receive_buffer (up to max_buffer_length).
```